### DISCUSSION TITLE
Error: Cannot find module

### DISCUSSION THREAD

**Discussion by brunodsazevedo** [adonisjs/core#2127] [Help] [answered]
I created an ApiRest application with AdonisJS v4.1, however I am having trouble creating an authentication controller using JWT configuration from the documentation.

This is my controller that I created with adonis make: controller auth

`'use strict'

const User = use('App/Models/Usuario');

class AuthController {

    async authentication ({request, response, auth}){

        const { email, password } = request.all();
        const user = await User.findBy('email', email);

        if(!user)
            return response.status(404).send({error: 'User not found'});

        if(user.status == 0)
            return response.status(403).send({error: 'User disable'});

        const token = await auth.attempt(email, password);

        return token;

    }

}

module.exports = AuthController`

And here are my routes:

`'use strict'

/*
|--------------------------------------------------------------------------
| Routes
|--------------------------------------------------------------------------
|
| Http routes are entry points to your web application. You can create
| routes for different URLs and bind Controller actions to them.
|
| A complete guide on routing is available here.
| http://adonisjs.com/docs/4.1/routing
|
*/

/** @type {typeof import('@adonisjs/framework/src/Route/Manager')} */
const Route = use('Route')

Route.get('/', () => {
  return { greeting: 'Hello world in JSON' }
})

Route.post('/login', 'AuthController.authentication');

Route.resource('usuario', 'UsuarioController').apiOnly();`

When I try to access the route / login I get this error:

`Error: Cannot find module 'C:\projetos\generApp\backend\app\Models\User'
Require stack:
- C:\projetos\generApp\backend\node_modules\require-stack\src\index.js
- C:\projetos\generApp\backend\node_modules\require-stack\index.js
- C:\projetos\generApp\backend\node_modules\@adonisjs\fold\src\Ioc\index.js
- C:\projetos\generApp\backend\node_modules\@adonisjs\fold\index.js
- C:\projetos\generApp\backend\server.js
Error: Cannot find module 'C:\projetos\generApp\backend\app\Models\User'
Require stack:
- C:\projetos\generApp\backend\node_modules\require-stack\src\index.js
- C:\projetos\generApp\backend\node_modules\require-stack\index.js
- C:\projetos\generApp\backend\node_modules\@adonisjs\fold\src\Ioc\index.js
- C:\projetos\generApp\backend\node_modules\@adonisjs\fold\index.js
- C:\projetos\generApp\backend\server.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at requireStack (C:\projetos\generApp\backend\node_modules\require-stack\src\index.js:44:12)
    at Ioc._resolveAutoloadedPath (C:\projetos\generApp\backend\node_modules\@adonisjs\fold\src\Ioc\index.js:265:20)
    at Ioc.make (C:\projetos\generApp\backend\node_modules\@adonisjs\fold\src\Ioc\index.js:815:40)
    at LucidSerializer.setConfig (C:\projetos\generApp\backend\node_modules\@adonisjs\auth\src\Serializers\Lucid.js:122:23)
    at Auth.authenticator (C:\projetos\generApp\backend\node_modules\@adonisjs\auth\src\Auth\index.js:113:24)
    at new Auth (C:\projetos\generApp\backend\node_modules\@adonisjs\auth\src\Auth\index.js:68:39)`

O que estou fazendo de errado?

**Comment by thetutlage** [comment]
Format your code blocks please. Here's the markdown guide https://guides.github.com/features/mastering-markdown/

**Reply by brunodsazevedo** [reply]
Sorry. Here it is block code.

this is my routes:

```

'use strict'

/*
|--------------------------------------------------------------------------
| Routes
|--------------------------------------------------------------------------
|
| Http routes are entry points to your web application. You can create
| routes for different URLs and bind Controller actions to them.
|
| A complete guide on routing is available here.
| http://adonisjs.com/docs/4.1/routing
|
*/

/** @type {typeof import('@adonisjs/framework/src/Route/Manager')} */
const Route = use('Route')

Route.get('/', () => {
  return { greeting: 'Hello world in JSON' }
})

Route.post('/login', 'AuthController.authentication');

Route.resource('usuario', 'UsuarioController').apiOnly();

```

my controller auth

```

'use strict'

const User = use('App/Models/Usuario');

class AuthController {

    async authentication ({request, response, auth}){

        const { email, password } = request.all();
        const user = await User.findBy('email', email);

        if(!user)
            return response.status(404).send({error: 'User not found'});

        if(user.status == 0)
            return response.status(403).send({error: 'User disable'});

        const token = await auth.attempt(email, password);

        return token;

    }

}

module.exports = AuthController
```

and error post method

```
Error: Cannot find module 'C:\projetos\generApp\backend\app\Models\User'
Require stack:

C:\projetos\generApp\backend\node_modules\require-stack\src\index.js
C:\projetos\generApp\backend\node_modules\require-stack\index.js
C:\projetos\generApp\backend\node_modules@adonisjs\fold\src\Ioc\index.js
C:\projetos\generApp\backend\node_modules@adonisjs\fold\index.js
C:\projetos\generApp\backend\server.js
Error: Cannot find module 'C:\projetos\generApp\backend\app\Models\User'
Require stack:
C:\projetos\generApp\backend\node_modules\require-stack\src\index.js
C:\projetos\generApp\backend\node_modules\require-stack\index.js
C:\projetos\generApp\backend\node_modules@adonisjs\fold\src\Ioc\index.js
C:\projetos\generApp\backend\node_modules@adonisjs\fold\index.js
C:\projetos\generApp\backend\server.js
at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
at Function.Module._load (internal/modules/cjs/loader.js:725:27)
at Module.require (internal/modules/cjs/loader.js:952:19)
at require (internal/modules/cjs/helpers.js:88:18)
at requireStack (C:\projetos\generApp\backend\node_modules\require-stack\src\index.js:44:12)
at Ioc._resolveAutoloadedPath (C:\projetos\generApp\backend\node_modules@adonisjs\fold\src\Ioc\index.js:265:20)
at Ioc.make (C:\projetos\generApp\backend\node_modules@adonisjs\fold\src\Ioc\index.js:815:40)
at LucidSerializer.setConfig (C:\projetos\generApp\backend\node_modules@adonisjs\auth\src\Serializers\Lucid.js:122:23)
at Auth.authenticator (C:\projetos\generApp\backend\node_modules@adonisjs\auth\src\Auth\index.js:113:24)
at new Auth (C:\projetos\generApp\backend\node_modules@adonisjs\auth\src\Auth\index.js:68:39)
```

**Comment by RomainLanz** [comment, marked as answer]
Since you are not using the standard naming, have you change the name of the `User` model inside `config/auth.js` to match yours?

**Reply by RomainLanz** [reply]
@brunodsazevedo I'm talking about this line: https://github.com/adonisjs/adonis-fullstack-app/blob/master/config/auth.js#L32

**Reply by brunodsazevedo** [reply]
> Since you are not using the standard naming, have you change the name of the `User` model inside `config/auth.js` to match yours?

There is the error. My model User is `Usuario`, because i am build aplicattion in portuguese.

Thanks for your help @RomainLanz @thetutlage

**Comment by bogordesaincom** [comment]
this error . 
```
Error: Cannot find module 'C:\projetos\generApp\backend\app\Models\User'
```
what model name?

**Reply by thetutlage** [reply]
<img width="669" alt="Screenshot 2021-01-15 at 9 33 07 AM" src="https://user-images.githubusercontent.com/1706381/104679907-ae4a4f00-5714-11eb-980a-4ba71ae88172.png">


This ^^
